application/helpers/remotecontrol/remotecontrol_handle.php

```diff
@@ -3881,6 +3883,18 @@ public function cpd_importParticipants($sSessionKey, $participants, $update = fa
             }
         }
 
+        // Remove param is set to True
+        // Remove all participants whose ID does not appear in the import list.
+        if ($remove) {
+            $allParticipants = Participant::model()->getParticipantsWithoutLimit();
```

Comment:
This might not scale. If nr of participants is too high, RAM will explode. Fetch by batch query or use other strategy.

Reply:
It's not related to this commit here. It's an existing `Participant::model()->getParticipantsWithoutLimit();` issue. And this function exist.

Reply:
(But a deleteAll($criteria) with criteria IN array can be cool

Reply:
Still, then it is not the correct function to use.

Reply:
> Still, then it is not the correct function to use.

Yes, and maybe cpd_removeParticipants new function is better, no ?

Reply:
How to move forward with this PR?

Reply:
Either write a custom query to apply the change without using the ActiveRecord system, or use the batch query system in Yii. Hm I'm only able to find batch query for Yii 1.1, not 2. Perhaps your google skills are better? Or ask on their Slack channel.

Reply:
Custom queries should use the query builder, btw, preferably not raw SQL (because of the different vendors we support).

Reply:
A deleteAlll can fix the issue … one call with a [NOT IN](https://www.yiiframework.com/doc/api/1.1/CDbCriteria#addNotInCondition-detail)

Reply:
Right, good one :)